appeals/web/src/server/appeals/appeal-details/withdrawal/withdrawal.controller.js

```diff
@@ -0,0 +1,369 @@
+		false,
+		`Upload appellant's withdrawal request`,
+		[],
+		false
+	);
+};
+
+/**
+ * @param {import('@pins/express/types/express.js').Request} request
+ * @param {import('@pins/express/types/express.js').RenderedResponse<any, any, Number>} response
+ */
+export const getDateWithdrawalRequest = async (request, response) => {
+	return renderDateWithdrawalRequest(request, response);
+};
+
+/**
+ * @param {import('@pins/express/types/express.js').Request} request
+ * @param {import('@pins/express/types/express.js').RenderedResponse<any, any, Number>} response
+ */
+export const postDateWithdrawalRequest = async (request, response) => {
+	const { currentAppeal } = request;
+	if (!currentAppeal || currentAppeal.appealStatus === 'withdrawn') {
+		return response.status(404).render('app/404');
+	}
+
+	try {
+		const { appealId } = request.params;
+		const {
+			'withdrawal-request-date-day': day,
+			'withdrawal-request-date-month': month,
+			'withdrawal-request-date-year': year
+		} = request.body;
+		const { errors } = request;
+
+		if (errors) {
+			return renderDateWithdrawalRequest(request, response);
+		}
+
+		const withdrawalRequestDate = new Date(year, month - 1, day);
+
+		/** @type {import('./withdrawal.types.js').WithdrawalRequest} */
+		request.session.withdrawal = {
+			...request.session.withdrawal,
+			withdrawalRequestDate
+		};
+
+		return response.redirect(
+			`/appeals-service/appeal-details/${appealId}/withdrawal/redaction-status`
+		);
+	} catch (error) {
+		logger.error(error);
+		return response.status(500).render('app/500.njk');
+	}
+};
+
+/**
+ *
+ * @param {import('@pins/express/types/express.js').Request} request
+ * @param {import('@pins/express/types/express.js').RenderedResponse<any, any, Number>} response
+ */
+const renderDateWithdrawalRequest = async (request, response) => {
+	const { errors, currentAppeal, session } = request;
+
+	if (!currentAppeal || currentAppeal.appealStatus === 'withdrawn') {
+		return response.status(404).render('app/404');
+	}
+
+	if (!objectContainsAllKeys(session, ['fileUploadInfo'])) {
+		return response.status(500).render('app/500.njk');
+	}
+
+	/** @type {import('./withdrawal.types.js').WithdrawalRequest} */
+	request.session.withdrawal = {
+		...session.withdrawal,
+		documentId: session.fileUploadInfo[0].GUID
+	};
+
+	let withdrawalRequestDay = request.body['withdrawal-request-date-day'];
+	let withdrawalRequestMonth = request.body['withdrawal-request-date-month'];
+	let withdrawalRequestYear = request.body['withdrawal-request-date-year'];
+
+	if (!withdrawalRequestDay || !withdrawalRequestMonth || !withdrawalRequestYear) {
+		if (request.session.withdrawal?.withdrawalRequestDate) {
+			const withdrawalRequestDate = new Date(request.session.withdrawal.withdrawalRequestDate);
+			withdrawalRequestDay = withdrawalRequestDate.getDate();
+			withdrawalRequestMonth = withdrawalRequestDate.getMonth() + 1; // Months are 0-indexed
```

Comment:
If you convert to a `DayMonthYear` using `dateToDayMonthYear` in `appeals/web/src/server/lib/dates.js` first, it will handle the month indexing for you

Reply:
done